km63/Milevska_Olha/laboratory1.sql

```diff
@@ -1,2 +1,146 @@
 -- LABORATORY WORK 1
+Tables
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+видаляти таблиці та робити запити з таблиць
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER milevska IDENTIFIED BY milevska64
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+
+ALTER USER milevska QUOTA 100M ON USERS;
+GRANT "CONECT" TO milevska;  
+
+GRANT INSERT ANY TABLE TO milevska;
+GRANT DROP ANY TABLE TO milevska;
+
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+На комп'ютері встановлено OS Windows.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE TABLE user(
+    user_name VARCHAR(20)
+);
+
+ALTER TABLE user
+ADD CONSTRAINT user_name_pk PRIMARY KEY (user_name);
+
+CREATE TABLE os(
+    os_name VARCHAR(20)
+); 
+
+ALTER TABLE os
+ADD CONSTRAINT os_name_pk PRIMARY KEY (os_name);    
+
+CREATE TABLE user_os(
+    user_name_fk VARCHAR(20)
+    os_name_fk VARCHAR(20)
+);
+
+ALTER TABLE user_os
+ADD CONSTRAINT user_os_pk PRIMARY KEY (user_name_fk,os_name_fk );    
+
+ALTER TABLE user_os
+ADD CONSTRAINT user_name_fk FOREIGN KEY (user_name_fk) REFERENCES TO (user_name);
+
+ALTER TABLE user_os
+ADD CONSTRAINT os_name_fk FOREIGN KEY (os_name_fk) REFERENCES TO (os_name);
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Як звуть покупців, що не купляли найдешевший товар.
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
```

Comment:
SELECT CUST_NAME
FROM CUSTOMERS, ORDERITEMS, ORDERS
WHERE ITEM_PRICE  NOT IN(
    SELECT MIN(ITEM_PRICE)
    FROM ORDERITEMS);